tests/integration.opts

```diff
@@ -1 +1,3 @@
+--require dotenv/config
+--require ./src/config
```

Comment:
I wouldn't use the .env file for integration tests. Instead each test or test group should set values in `process.env` as needed and restore them when the test is complete.

Reply:
Thanks, @naptowncode. We have a question... why would we need to reset them? Won't process.env be reinstantiated for each test fixture? We just don't know.

Good call - we'll not use .env for the UTs. We plan to leave it in for the integration tests, though.

Reply:
I honestly wouldn't use .env even for the integration tests. People are going to muck around with that file to change settings in their local dev app. You don't want that to affect tests too.

All the tests run by a single execution of mocha will use the same process space, including the same env vars. But if you set env vars for one test you don't really want that bleeding over into unrelated tests. The cleanest way to handle it is with an `afterEach()` hook:

```javascript
const { env } = process;

describe('some tests', () => {
  beforeEach(() => {
    process.env = { ...env };
  });

  afterEach(() => {
    process.env = env;
  });

  it('changes env vars', () => {
    process.env.NODE_ENV = 'potato';
  });

  it('retains the original env vars', () => {
    assert(process.env.NODE_ENV === 'test');
  });
});
```

Reply:
Thanks, @naptowncode, but I'm nervous that by doing this to the integration tests will have the potential to cover up problems with a misconfigured environment.

We were thinking of using this approach for integration tests: https://devcenter.heroku.com/articles/heroku-ci#configuration-using-app-json

Reply:
I agree with @naptowncode.  I have a situation where I am doing development and unit testing on the same server, but using different MongoDb databases.  It makes sense to isolate setting the unit test environment variables to the tests scripts.

Reply:
What are your tests testing? If you want to test that it does what you expect when env vars are set a certain way, then test that. I don't understand how CI tests are supposed to catch a misconfigured environment.

Reply:
Any CI test, whether unit or integration or end-to-end, can only validate that your code does what you expect it to do. If the VAULT_API_KEY env var is required for the system to function then make it required in code and throw an error if it isn't specified. Then write a test where that var isn't specified and verify that the code throws the error like you expect.

This is why it's also a good idea to bring in a code coverage tool like nyc that can show you which lines and branches aren't getting executed by your tests.

Reply:
@naptowncode - Thanks for your input to this project. We made all of the changes you recommended, with the exception of hardcoding/faking the env vars for integration tests. (The guys on the team, who have approved this, agree that we don't want things like DB connections, API keys, etc. hardcoded in the *integration* test scripts.) As you recommend, our code is ensuring that required env vars are present. Can you take another quick look at this and approve it (or at least remove your rejection)?

Reply:
Sorry, different workflow than on my usual projects where "request changes" doesn't block outright. Didn't mean to hold up the show, my bad.

Reply:
Thanks! Good discussion, and work, everyone.